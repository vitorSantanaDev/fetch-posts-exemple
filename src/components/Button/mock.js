export const buttonProps = {
  text: 'Load More Posts',
  onClick: jest.fn(),
  disabled: false,
};
