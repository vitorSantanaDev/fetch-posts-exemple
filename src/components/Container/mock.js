import React from 'react';

export const propsContainer = {
  children: <h1>Children</h1>,
};
