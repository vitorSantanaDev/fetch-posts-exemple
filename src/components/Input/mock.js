export const propsInput = {
  value: '',
  onChange: jest.fn(),
};
