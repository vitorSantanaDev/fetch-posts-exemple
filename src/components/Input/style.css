.textInput {
  width: 100%;
  font-size: 32px;
  padding: 8px 16px;
  border-radius: 5px;
  border: none;
  outline: none;
  transition: ease-in-out .3s;
}

.textInput:hover, .textInput:focus {
  border: 1px solid #fea;
}