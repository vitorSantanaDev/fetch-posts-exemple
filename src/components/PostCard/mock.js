export const postCardProps = {
  title: 'title one',
  body: 'Body one',
  cover: 'img/img.png',
};
