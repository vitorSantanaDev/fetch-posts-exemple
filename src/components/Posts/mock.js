export const postsProps = [
  { id: 1, title: 'Title Post one', body: 'Body text Post one', cover: '/image/image.png' },
  { id: 2, title: 'Title Post two', body: 'Body text Post two', cover: '/image/image.png' },
  { id: 3, title: 'Title Post three', body: 'Body text Post three', cover: '/image/image.png' },
];
