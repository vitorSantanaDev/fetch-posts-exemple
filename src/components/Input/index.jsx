import React from 'react';
import Types from 'prop-types';
import './style.css';

const Input = ({ value, onChange }) => {
  return <input placeholder="Type your search" className="textInput" type="search" value={value} onChange={onChange} />;
};

Input.propTypes = {
  value: Types.string.isRequired,
  onChange: Types.func,
};

export default Input;
