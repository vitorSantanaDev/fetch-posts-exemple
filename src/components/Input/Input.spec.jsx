import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Input from '.';
import { propsInput } from './mock';

describe('<Input />', () => {
  it('should have value of value prop', () => {
    render(<Input onChange={propsInput.onChange} value={propsInput.value} />);
    const input = screen.getByPlaceholderText(/type your search/i);

    expect(input.value).toBe('');
  });

  it('should call onChange function each key pressed', () => {
    const value = 'New value';
    render(<Input onChange={propsInput.onChange} value={value} />);
    const input = screen.getByPlaceholderText(/type your search/i);

    userEvent.type(input, value);
    expect(input.value).toBe(value);
    expect(propsInput.onChange).toHaveBeenCalledTimes(9);
  });

  it('should match snapshot', () => {
    const { container } = render(<Input onChange={propsInput.onChange} value={propsInput.value} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
