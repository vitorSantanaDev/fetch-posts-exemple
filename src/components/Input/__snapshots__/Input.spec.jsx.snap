// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match snapshot 1`] = `
<input
  class="textInput"
  placeholder="Type your search"
  type="search"
  value=""
/>
`;
